src/runtime/io.cpp

```diff
@@ -521,6 +524,194 @@ extern "C" LEAN_EXPORT obj_res lean_io_prim_handle_put_str(b_obj_arg h, b_obj_ar
+extern "C" LEAN_EXPORT obj_res lean_get_current_time(obj_arg /* w */) {
+    using namespace std::chrono;
+
+    std::chrono::system_clock::time_point now = std::chrono::system_clock::now();
+    long long timestamp = std::chrono::duration_cast<std::chrono::nanoseconds>(now.time_since_epoch()).count();
+
+    long long secs = timestamp / 1000000000;
+    long long nano = timestamp % 1000000000;
+
+    lean_object *lean_ts = lean_alloc_ctor(0, 2, 0);
+    lean_ctor_set(lean_ts, 0, lean_int_to_int(static_cast<int>(secs)));
+    lean_ctor_set(lean_ts, 1, lean_int_to_int(static_cast<int>(nano)));
+
+    return lean_io_result_mk_ok(lean_ts);
+}
+
+/* Std.Time.TimeZone.getWindowsTimeZoneAt : String -> UInt64 -> IO Timezone */
+extern "C" LEAN_EXPORT obj_res lean_get_windows_timezone_at(obj_arg timezone_str, uint64_t timestamp_secs, obj_arg /* w */) {
+#if defined(LEAN_WINDOWS)
+    UErrorCode status = U_ZERO_ERROR;
+    const char* dst_name = lean_string_cstr(timezone_str);
+
+    UChar tzID[256];
+    u_strFromUTF8(tzID, sizeof(tzID) / sizeof(tzID[0]), NULL, dst_name, strlen(dst_name), &status);
+
+    if (U_FAILURE(status)) {
+        return lean_io_result_mk_error(lean_decode_io_error(EINVAL, mk_string("failed to read identifier")));
+    }
+
+    UCalendar *cal = ucal_open(tzID, -1, NULL, UCAL_GREGORIAN, &status);
+
+    if (U_FAILURE(status)) {
+        ucal_close(cal);
+        return lean_io_result_mk_error(lean_decode_io_error(EINVAL, mk_string("failed to open calendar")));
+    }
+
+    ucal_setMillis(cal, timestamp_secs * 1000, &status);
+    if (U_FAILURE(status)) {
+        ucal_close(cal);
+        return lean_io_result_mk_error(lean_decode_io_error(EINVAL, mk_string("failed to set calendar time")));
+    }
+
+    UChar display_name[32];
+    int32_t display_name_len = ucal_getTimeZoneDisplayName(cal, UCAL_SHORT_STANDARD, "en_US", display_name, 32, &status);
+
+    if (U_FAILURE(status)) {
+        ucal_close(cal);
+        return lean_io_result_mk_error(lean_decode_io_error(EINVAL, mk_string("failed to read abbreaviation")));
+    }
+
+    char display_name_str[256];
+    u_strToUTF8(display_name_str, sizeof(display_name_str), NULL, display_name, display_name_len, &status);
+
+    if (U_FAILURE(status)) {
+        ucal_close(cal);
+        return lean_io_result_mk_error(lean_decode_io_error(EINVAL, mk_string("failed to get abbreviation to cstr")));
+    }
+
+    int32_t zone_offset = ucal_get(cal, UCAL_ZONE_OFFSET, &status);
+
+    if (U_FAILURE(status)) {
+        ucal_close(cal);
+        return lean_io_result_mk_error(lean_decode_io_error(EINVAL, mk_string("failed to get zone_offset")));
+    }
+
+    int32_t dst_offset = ucal_get(cal, UCAL_DST_OFFSET, &status);
+
+    zone_offset += dst_offset;
+    
+    if (U_FAILURE(status)) {
+        return lean_io_result_mk_error(lean_decode_io_error(EINVAL, mk_string("failed to get dst_offset")));
+    }
+
+    int offset_hour = zone_offset / 3600000;
```

Comment:
Looks like you're leaking the calendar object here. I was able to reproduce the leak using
```lean
import Std.Time

open Std.Time

def main (_ : List String) : IO Unit := do
  for _ in [0:100000000] do
    discard Database.defaultGetCurrentTimeZone
```
(though that calls the other function below I think). Running this using `lean --run`, I can see a steady memory increase in the Windows task manager.

Also, the ICU examples at https://unicode-org.github.io/icu/userguide/datetime/calendar/examples.html don't close the calendar in case of an error. Could you double-check whether calendars should be closed on error?